src/core/kconfig.cpp

```diff
@@ -591,7 +590,151 @@ QString KConfig::mainConfigName()
     QString appName = QCoreApplication::applicationName();
     return appName + QLatin1String("rc");
 }
+#else
+MainConfigInformation KConfig::mainConfigName()
```

Comment:
I keep the old implementations in case the library is used without elektra. But the functionality can be merged, after I have checked, if it gives the same results as the old implementation. (No Elektra code is used in this method &rarr; can be used even if Elektra is not present)

Reply:
You might be right: it might be easier to get this merged if the PR does not change anything if ELEKTRA is turned off. But is this still the case?